modules/material/themes/material/default/selectidp-links.twig

```diff
@@ -0,0 +1,129 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>{{ '{selectidp-links:title}'|trans }}</title>
+
+  {% include 'header.twig' %}
+
+  <script>
+    function setSelectedIdp(id) {
+      const idpInput = document.createElement('input');
+
+      idpInput.type = 'hidden';
+      idpInput.name = '{{ returnIDParam|e('js')|raw }}';
+      idpInput.value = id;
+
+      document.querySelector('form').appendChild(idpInput);
+
+      ga('send', 'event', 'hub', 'choice', 'IdP', id);
+    }
+
+    function clickedAnyway(idpName) {
+      ga('send', 'event', 'hub', 'choice-disabled', 'IdP', idpName);
+    }
+  </script>
+</head>
+<body>
+<div class="mdl-layout mdl-layout--fixed-header fill-viewport">
+  <header class="mdl-layout__header">
+    <div class="mdl-layout__header-row">
+      <span class="mdl-layout-title scale-to-parent">
+        {% if spName ?? '' is empty %}
+          {{ '{selectidp-links:header}'|trans }}
+        {%  else %}
+          {{ '{selectidp-links:header-for-sp}'|trans({'%spName%': spName|e}) }}
+        {% endif %}
+      </span>
+
+      <div class="mdl-layout-spacer"></div>
+
+      {% if not helpCenterUrl ?? '' is empty %}
+        <nav class="mdl-navigation">
+          <a href="{{ helpCenterUrl|e('html_attr') }}" target="_blank" class="mdl-navigation__link">
+            {{ '{selectidp-links:help}'|trans }}
+          </a>
+        </nav>
+      {% endif %}
+    </div>
+  </header>
+
+  <main class="mdl-layout__content">
+    {% include 'announcement.twig' %}
+
+    <form layout-children="row" child-spacing="space-around">
+      <input type="hidden" name="entityID" value="{{ entityID|e('html_attr') }}"/>
+      <input type="hidden" name="return" value="{{ return|e('html_attr') }}"/>
+      <input type="hidden" name="returnIDParam" value="{{ returnIDParam|e('html_attr') }}"/>
+
+      {% for idp in idplist %}
+        <div
+          class="mdl-card mdl-shadow--8dp row-aware"
+          title="{{ '{selectidp-links:enabled}'|trans({'%idpName%': idp.name|e}) }}"
+        >
+          <div class="mdl-card__media white-bg fixed-height">
+            <button
+              class="mdl-button logo-container fill-parent"
+              onclick="setSelectedIdp('{{ idp.entityid|e('js')|raw }}')"
```

Comment:
probably not a necessary improvement at this point but I've been wondering if there is a way to get away from inline event handlers. See [mdn says don't use these](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#inline_event_handlers_%E2%80%94_dont_use_these)

Reply:
Would it be worth its own YT issue?

Reply:
Probably. I can create one if you like.

Reply:
Here we go [IDP-1123](https://itse.youtrack.cloud/issue/IDP-1123)

Reply:
Thanks!